origin_ui/register_origin.go

```diff
@@ -0,0 +1,200 @@
+/***************************************************************
+ *
+ * Copyright (C) 2023, Pelican Project, Morgridge Institute for Research
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ***************************************************************/
+
+package origin_ui
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/pelicanplatform/pelican/config"
+	"github.com/pelicanplatform/pelican/metrics"
+	nsregistry "github.com/pelicanplatform/pelican/namespace-registry"
+	"github.com/pelicanplatform/pelican/param"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type (
+	errorResp struct {
+		Error string `json:"error"`
+	}
+
+	keyStatus int
+)
+
+const (
+	noKeyPresent keyStatus = iota
+	keyMismatch
+	keyMatch
+)
+
+func keyIsRegistered(privkey jwk.Key, url string) (keyStatus, error) {
+	keyId := privkey.KeyID()
+	if keyId == "" {
+		return noKeyPresent, errors.New("Provided key is missing a key ID")
+	}
+	key, err := privkey.PublicKey()
+	if err != nil {
+		return noKeyPresent, err
+	}
+
+	req, err := http.NewRequest("GET", url+"/.well-known/issuer.jwks", nil)
+	if err != nil {
+		return noKeyPresent, err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	tr := config.GetTransport()
+	client := &http.Client{Transport: tr}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return noKeyPresent, err
+	}
+	defer resp.Body.Close()
+
+	// Check HTTP response -- should be 200, else something went wrong
+	body, _ := io.ReadAll(resp.Body)
+	if resp.StatusCode == 404 || resp.StatusCode == 500 {
```

Comment:
@bbockelm I think what I saw corresponds to what you put in the TODO here. The 500 is just a false alarm and I agreed we should tackle this in the future when users get used to the auto-register flow and we don't want to bother registry admin to saw a bunch of 500 logs that are actually expected "attempts". Hopefully this makes sense!